libkbfs/block_retrieval_queue.go

```diff
@@ -455,7 +559,25 @@ func (brq *blockRetrievalQueue) FinalizeRequest(
 		// Since we created this channel with a buffer size of 1, this won't
 		// block.
 		req.doneCh <- err
+		doneChans = append(doneChans, r.prefetchDoneCh)
+		errChans = append(errChans, r.prefetchErrCh)
 	}
+
+	go func() {
+		// Fan-out the prefetch result to the channel of each individual
+		// request. Since the slice was created before the request mutex was
```

Comment:
Above there's a comment that says the request mutex is symbolic, but that seems like it's not the case anymore.

Reply:
`brq.mtx` protects the overall queue. As soon as we've deleted this pointer from `brq.ptrs` (protected by `brq.mtx`), nobody else can interfere with this request. However, `retrieval.requests` is technically unprotected by a mutex after the `brq.mtx` unlock, which I found flakily tripping the race detector (an unprotected read where there were protected writes). IMO this is actually a bug in the race detector, but I addressed it by protecting this particular resource with its own mutex.

Reply:
I'm just arguing that we should remove the word "symbolic" above if the lock is important in some way (even for the race detector), because otherwise some future person might just decide to remove it.